0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,408 @@
+- `triggeredBy`: `peer.settle`
+- `message`: _Determined by connector_
+- `data`: _Response message from recipient settlement engine, or empty if antecedent failure_
+
+### Motivation
+
+Settlement engines supercede the [Ledger Plugin Interface (LPIv2)](../deprecated/0024-ledger-plugin-interface-2/0024-ledger-plugin-interface-2.md), an earlier abstraction for settlement integrations with Interledger. This new model addresses these issues:
+
+1. Multi-account plugins required logic for handling ILP packets, increasing implementation complexity.
+2. Plugins bundled settlement and bilateral communication functionality together, limiting composability.
+3. JavaScript plugins limited interoperability with non-JavaScript connector implementations.
+4. Plugins operated in the same process as the connector, which limited scaling the two services independently.
+
+## Specification
+
+### Accounts and identifiers
+
+Each account MUST be identified by a unique, [URL-safe](https://tools.ietf.org/html/rfc3986#section-2.3) string.
+
+The settlement engine MUST be responsible for correlating an account identifier to the peer's identity on the shared ledger or settlement system, if required. For separation of concerns between clearing and settlement, the accounting system is NOT RECOMMENDED to have knowledge of the peer's identity on the shared settlement system.
+
+### Units and quantities
+
+An amount of a given asset or currency may be represented using different denominations or units of the asset. For example, one U.S. dollar may be represented as \$1 or 100 cents, and one Bitcoin may be represented as 1 BTC or 100000000 satoshis.
+
+A **standard unit** is the typical unit of value for a particular asset, such as \$1 in the case of U.S. dollars, or 1 BTC in the case of Bitcoin.
+
+A **fractional unit** represents some unit smaller than the standard unit, but with greater precision. Examples of fractional monetary units include one cent (\$0.01 USD), or 1 satoshi (0.00000001 BTC).
+
+An **asset scale** is the difference in orders of magnitude between a **standard unit** and a corresponding **fractional unit**. More formally, the asset scale is a non-negative integer (0, 1, 2, …) such that one **standard unit** equals `10^(-scale)` of a corresponding **fractional unit**. If the fractional unit equals the standard unit, then the asset scale is 0.
+
+For example, one cent represents an asset scale of 2 in the case of USD, whereas 1 satoshi represents an asset scale of 8 in the case of Bitcoin.
+
+#### `Quantity` JSON type
+
+A quantity represents an amount denominated in some unit of a particular fungible asset. (Since account balances may only be denominated in a single asset, the type of asset should be implicit).
+
+##### Attributes
+
+- **`amount`** string
+  - Number of the given unit of the asset between 0 and the maximum 64-bit unsigned integer, 18,446,744,073,709,551,615 (inclusive).
```

Comment:
@kincaidoneil Given our discussion this morning, is this value no longer constrained? In other words, shouldn't it be allowable for the API calls to be made with BigInteger-type numbers?

Reply:
Good catch! For now, changed it to "non-negative integer." (Should we have any upper bound?)

Reply:
hmmm...it's probably fine as-is. I think we all agreed that the APIs would need to support BigNumbers. In Java at least a `BigInteger` is generally only limited by system memory.